mlflow/pipelines/steps/train.py

```diff
@@ -608,9 +659,25 @@ def objective(hyperparameter_args):
         hardcoded_estimator_loss = objective(estimator_hardcoded_params)
 
         if best_hp_estimator_loss < hardcoded_estimator_loss:
-            best_combined_params = dict(estimator_hardcoded_params, **best_hp_params)
+            best_hardcoded_params = {
+                param_name: param_value
+                for param_name, param_value in estimator_hardcoded_params.items()
+                if param_name not in best_hp_params
+            }
         else:
-            best_combined_params = estimator_hardcoded_params
+            best_hp_params = {}
+            best_hardcoded_params = estimator_hardcoded_params
+
+        best_combined_params = dict(estimator_hardcoded_params, **best_hp_params)
+
+        self._write_yaml(
+            root=output_directory, file_name="best_hyperparameters.yaml", data=best_hp_params
+        )
+        self._write_yaml(
+            root=output_directory,
+            file_name="best_hardcoded_parameters.yaml",
+            data=best_hardcoded_params,
+        )
```

Comment:
Apologies, when we discussed displaying the best parameters on the step card in separate cells / divs of the same tab last week, I didn't think about the YAML artifact output behavior. IMO, we should still write parameters to a single file where tuned parameters are listed first, followed by hardcoded parameters. We should separate these sections by a comment, perhaps by making one `yaml.safe_dump()` call, writing a comment at the end of the file, and then making a second `yaml.safe_dump()` call where the second calls *appends* to the file

Reply:
Can we also make `best_parameters.yaml` accessible via `Pipeline().get_artifact("best_parameters")`?